apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -52,26 +53,80 @@ class AnimationPickerBody extends React.Component {
     spriteLab: PropTypes.bool
   };
 
-  state = {
-    searchQuery: '',
-    categoryQuery: '',
-    currentPage: 0
+  constructor(props) {
+    super(props);
+
+    const initialState = {
+      searchQuery: '',
+      categoryQuery: '',
+      currentPage: 0,
+      libraryManifest: props.spriteLab ? spriteCostumeLibrary : animationLibrary
+    };
+    const results = this.searchAssets(initialState.currentPage, initialState);
+    this.state = {
+      ...initialState,
+      results
+    };
+  }
+
+  searchAssets = (page, config = {}) => {
+    let {searchQuery, categoryQuery, libraryManifest} = config;
+
+    // Set defaults if any config values are undefined.
+    if (searchQuery === undefined) {
+      searchQuery = this.state.searchQuery;
+    }
+    if (categoryQuery === undefined) {
+      categoryQuery = this.state.categoryQuery;
+    }
+    if (libraryManifest === undefined) {
+      libraryManifest = this.state.libraryManifest;
+    }
+
+    const {results} = searchAssets(
+      searchQuery,
+      categoryQuery,
+      libraryManifest,
+      page,
+      MAX_SEARCH_RESULTS
+    );
+    return results;
   };
 
-  onSearchQueryChange = value => {
-    this.setState({searchQuery: value, currentPage: 0});
+  handleScroll = event => {
+    const scrollWindow = event.target;
+    if (
+      scrollWindow.scrollTop + MAX_HEIGHT >=
+      scrollWindow.scrollHeight * 0.9
+    ) {
+      const nextPage = this.state.currentPage + 1;
```

Comment:
is it possible that nextPage would keep increasing indefinitely? i.e. when I get to the bottom of the results?

Reply:
yeah, `nextPage` would continue to increment, but only until you reach the end of the modal (since this function runs on scroll). it has no effect since there are no more results, but i can add an exit state to clean this up (i.e., if there are no more results, set `doneLoadingResults` to true in state and don't attempt to load any new pages/results)